medusa/providers/torrent/html/btdb.py

```diff
@@ -0,0 +1,132 @@
+    """BTDB Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(BTDBProvider, self).__init__('BTDB')
+
+        # Credentials
+        self.public = True
+
+        # URLs
+        self.url = 'https://btdb.to'
+        self.urls = {
+            'search': urljoin(self.url, '/q/{query}/{page}?sort=popular'),
+        }
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=20)
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                for page in range(1, 4):
+                    search_url = self.urls['search'].format(query=search_string, page=page)
+
+                    response = self.session.get(search_url)
+                    if not response or not response.text:
+                        log.debug('No data returned from provider')
+                        continue
+
+                    results += self.parse(response.text, mode)
+
+        return results
+
+    def parse(self, data, mode):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+
+        :return: A list of items found
+        """
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            table_body = html.find('ul', {'class': 'search-ret-list'})
+
+            # Continue only if at least one release is found
+            if not table_body:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            for row in table_body.findAll('li', {'class': 'search-ret-item'}):
+                try:
+
+                    title = row.find('h2').find('a').get('title')
+                    download_url = row.find('div').find('a').get('href')
+                    if not all([title, download_url]):
+                        continue
+
+                    spans = row.find('div').findAll('span')
+
+                    torrent_size = spans[0].get_text()
+                    size = convert_size(torrent_size) or -1
+
+                    torrent_pubdate = spans[2].get_text()
+                    pubdate = self.parse_pubdate(torrent_pubdate)
+
+                    torrent_popularity = spans[3].get_text()
+                    seeders = leechers = int(torrent_popularity)
```

Comment:
This doesn't seem to have anything to do with seeders/leechers? If it's not available just hardcode `seeders = 1` and `leechers = 0`. Please also move it before `torrent_size`.

Reply:
It's related in some way. I mean, the more popularity, more seeders/leechers seen in the dht network.
This is useful to sort torrents in manual search. If the popularity is low maybe the torrent is dead.

Reply:
I see what you mean. But what if someone wants to download the torrent with the highest seeders (with multiple providers)? The torrents coming from BTDB will be almost always on top, although they might not have the highest amount of seeders. Is there any way to know how that popularity is calculated?

Reply:
Yes please, try to normalize the popularity so that it can be useful for us.